File: 9_abstraction.py
from abc import ABC, abstractmethod

class User(ABC):
    def __init__(self, name, email, age) -> None:
        self.name = name
        self.email = email 
        self.age = age
    
    @abstractmethod
    def displayData(self): # abstractmethod , ei user class ta jei jei class inherit korbe sei class er moddhe obossoi displayData method ta thakte i hobe
        raise NotImplementedError
    
class Person(User):
    def __init__(self, name, email, age) -> None:
        User.__init__(self, name, email, age)

    def displayData(self):
        print(self.name, self.email, self.age)


obj = Person("rahim", "rahim@gmail.com", 25)
obj.displayData()

File: 9_abstraction.cpp
#include<bits/stdc++.h>
using namespace std;
class User
{
    public:
        string name;
        string email;
        int age;

        User(string name, string email, int age)
        {
            this->name = name;
            this->email = email;
            this->age = age;
        }
        //pure virtual function er kono body thake na ty eke abstract method bola hoi
        virtual void DisplayData() = 0;
};
class Person : public User
{
    public:
        Person(string name, string email, int age) : User(name, email, age)
        {

        }
        // abstract method ke override kortaca DisplayData()
        void DisplayData()
        {
            cout<<name<<" "<<email<<" "<<age<<endl;
        }
};
int main()
{
    Person obj("rahim", "rahim@gmail.com", 25);
    obj.DisplayData();

    return 0;
}